src/binary-search-tree.js

```diff
@@ -31,8 +64,18 @@ class BinarySearchTree {
   // You'll need the queue-helper file for this. Or could you roll your own queue
   // again. Whatever floats your boat.
   breadthFirstForEach(cb) {
-
-  }
+    const current = [node];
+    
+        while (current.length > 0) {
```

Comment:
This is an interesting way you're going about solving this. I think it could work, but some of the details of your implementation are not quite right.